File: .skills/openclaw-skills/skills/mybusd/sol-bsc-dev-monitor/billing.js
// ═════════════════════════════════════════════════════════════
// SkillPay Billing Integration
// スキルペイ決統合 / 결제 연동 / Интеграция биллинга SkillPay
// ═══════════════════════════════════════════════════════════════════

const axios = require('axios');

// Billing API URL / 課金API URL / Ссылка на API биллинга
const BILLING_API_URL = 'https://skillpay.me/api/v1/billing';
const API_KEY = 'sk_f072a786149bc07fc8730b4683dc00f3e050e72441922284ca803cdee2b994b5';
const SKILL_ID = '282279e4-5370-4b9e-b5e7-9e07f0b3dc5c';

// Headers / ヘッダー / Заголовки
const headers = {
  'X-API-Key': API_KEY,
  'Content-Type': 'application/json'
};

// Check balance / 残高確認 / Проверка баланса
// Charge per call / 扣费 / 每次调用扣费 / 呼び出しとの課金 / Списание за вызов
async function chargeUser(userId, amount = 0.005) {
  try {
    const resp = await fetch(`${BILLING_API_URL}/charge`, {
      method: 'POST',
      headers: {
        'X-API-Key': API_KEY,
        'Content-Type': 'application/json'
      },
      body: JSON.stringify({
        user_id: userId,
        skill_id: SKILL_ID,
        amount
      })
    });
    
    const data = await resp.json();
    
    if (data.success) {
      return { ok: true, balance: data.balance };
    }
    
    return { 
      ok: false, 
      balance: data.balance, 
      payment_url: data.payment_url 
    };
  } catch (error) {
    console.error('Billing error:', error.message);
    return { ok: false, error: error.message };
  }
}

// Charge per call / 呼び出し / Списание за вызов
async function chargeUser(userId, amount = 0.005) {
  try {
    const resp = await fetch(`${BILLING_API_URL}/charge`, {
      method: 'POST',
      headers: headers,
      body: JSON.stringify({
        user_id: userId,
        skill_id: SKILL_ID,
        amount: amount
      })
    });
    const data = await resp.json();
    return {
      ok: data.success,
      balance: data.balance,
      paymentUrl: data.payment_url
    };
  } catch (error) {
    console.error('Charge error:', error.message);
    return {
      ok: false,
      balance: 0,
      error: error.message
    };
  }
}

// Generate payment link / 入金リンク生成 / Создание ссылки на оплату
async function getPaymentLink(userId, amount = 8) {
  try {
    const resp = await fetch(`${BILLING_API_URL}/payment-link`, {
      method: 'POST',
      headers: headers,
      body: JSON.stringify({
        user_id: userId,
        amount: amount
      })
    });
    const data = await resp.json();
    return data.payment_url;
  } catch (error) {
    console.error('Payment link error:', error.message);
    return null;
  }
}

module.exports = {
  checkBalance,
  chargeUser,
  getPaymentLink
};

File: .skills/openclaw-skills/skills/billjamno58/tax-invoice-validator/scripts/billing.py
# ═══════════════════════════════════════════════════
# SkillPay Billing Integration
# Pay-per-call: $0.01 USDT per analysis run
# ═══════════════════════════════════════════════════
import os
import requests

BILLING_API_URL = "https://skillpay.me"
BILLING_API_KEY = os.environ.get("SKILL_BILLING_API_KEY", "")
SKILL_ID = os.environ.get("SKILL_BILLING_SKILL_ID", "tax-invoice-validator")
HEADERS = {"X-API-Key": BILLING_API_KEY, "Content-Type": "application/json"}


def check_balance(user_id: str) -> float:
    """Check user's USDT balance."""
    if not BILLING_API_KEY:
        return float("inf")  # Dev mode: no billing
    resp = requests.get(
        f"{BILLING_API_URL}/api/v1/billing/balance",
        params={"user_id": user_id},
        headers=HEADERS,
        timeout=10,
    )
    return resp.json().get("balance", 0.0)


def charge_user(user_id: str, amount: float = 0.01) -> dict:
    """Charge user per call. Returns dict with ok status and balance."""
    if not BILLING_API_KEY:
        return {"ok": True, "balance": 999.0}  # Dev mode
    resp = requests.post(
        f"{BILLING_API_URL}/api/v1/billing/charge",
        headers=HEADERS,
        json={"user_id": user_id, "skill_id": SKILL_ID, "amount": amount},
        timeout=10,
    )
    data = resp.json()
    if data.get("success"):
        return {"ok": True, "balance": data.get("balance", 0.0)}
    return {
        "ok": False,
        "balance": data.get("balance", 0.0),
        "payment_url": data.get("payment_url"),
    }


def get_payment_link(user_id: str, amount: float = 10.0) -> str:
    """Generate payment link for user to top up."""
    if not BILLING_API_KEY:
        return ""
    resp = requests.post(
        f"{BILLING_API_URL}/api/v1/billing/payment-link",
        headers=HEADERS,
        json={"user_id": user_id, "amount": amount},
        timeout=10,
    )
    return resp.json().get("payment_url", "")
